File: app/src/main/java/com/lumiyaviewer/lumiya/dao/CachedResponse.java
package com.lumiyaviewer.lumiya.dao;

public class CachedResponse {
    private byte[] data;
    private String key;
    private boolean mustRevalidate;

    public CachedResponse() {
    }

    public CachedResponse(String str) {
        this.key = str;
    }

    public CachedResponse(String str, byte[] bArr, boolean z) {
        this.key = str;
        this.data = bArr;
        this.mustRevalidate = z;
    }

    public byte[] getData() {
        return this.data;
    }

    public String getKey() {
        return this.key;
    }

    public boolean getMustRevalidate() {
        return this.mustRevalidate;
    }

    public void setData(byte[] bArr) {
        this.data = bArr;
    }

    public void setKey(String str) {
        this.key = str;
    }

    public void setMustRevalidate(boolean z) {
        this.mustRevalidate = z;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/dao/CachedResponse.kt
package com.lumiyaviewer.lumiya.dao

class CachedResponse {
    private var data: ByteArray? = null
    private var key: String? = null
    private var mustRevalidate: Boolean = false

    constructor()

    constructor(key: String?) {
        this.key = key
    }

    constructor(key: String?, bytes: ByteArray?, mustRevalidate: Boolean) {
        this.key = key
        this.data = bytes
        this.mustRevalidate = mustRevalidate
    }

    fun getData(): ByteArray? = data

    fun getKey(): String? = key

    fun getMustRevalidate(): Boolean = mustRevalidate

    fun setData(bytes: ByteArray?) {
        this.data = bytes
    }

    fun setKey(key: String?) {
        this.key = key
    }

    fun setMustRevalidate(mustRevalidate: Boolean) {
        this.mustRevalidate = mustRevalidate
    }
}
